writeengine/wrapper/writeengine.cpp

```diff
@@ -837,84 +837,23 @@ int WriteEngineWrapper::deleteRow(const TxnID& txnid, const vector<CSCTypesList>
 inline void allocateValArray(void*& valArray, ColTupleList::size_type totalRow,
                              ColType colType, int colWidth)
 {
+    // All three cases effectively uses colWidth however left the switch
+    // for readability and potential Token size changes.
```

Comment:
This statement is not necessarily true. For a CHAR field, `colWidth` is not always equal to `MAX_COLUMN_BOUNDARY`. If we replace the entire switch block with a single line `valArray = calloc(totalRow, colWidth);` , we get a crash in test000.

Reply:
Really. Could you tell me where exactly b/c I had no crashes running test000 with deb builds prior the David's WF patch.

Reply:
The crash was in mysql/scripts/create_tpch_4byte.sql while running test000 when we just used `calloc(totalRow, colWidth)`. Specifically `create table orders` statement.